pyvrp/stop/MaxIterations.py

```diff
@@ -1,4 +1,4 @@
-from pyvrp.OptimisationTarget import OptimisationTarget
+from typing import Union
```

Comment:
Here, and in the other cases: just use `float`. MyPy treats `int` as a special case of `float` for type checking, so there's no real need for this amount of formality.

Reply:
The `OptimisationTarget` was using the `Union` which is why I kept it like this but I'll change it.